index.js

```diff
@@ -0,0 +1,69 @@
+export default function createMap(equals) {
+	const map = new Array(256)
+	return {
+		get: (key) => lookup(equals, map, key, 0),
+		set: (key, value) => set(equals, map, key, 0, value),
+		delete: (key) => remove(equals, map, key, 0),
+		getReference: (key, insert = false) =>
+			getReference(equals, map, key, 0, insert),
+	}
+}
+
+function set(equals, map, key, keyOffset, value) {
+	const k = key[keyOffset]
+	const r = map[k]
+	if (Array.isArray(r)) {
+		return set(equals, r, key, keyOffset + 1, value)
```

Comment:
To prevent `RangeError: Maximum call stack size exceeded` for large keys(~10k), you can replace recursion with 'trampoline pattern'
https://hackernoon.com/on-recursion-and-trampolining-uxf3tvx

Reply:
Did you tricker this with uniformly distributed keys?

Reply:
you mean "trigger this"?
Yes, but only with very large keys > 10_000 bytes

For anything smaller is the recursion faster

Reply:
> you mean "trigger this"?

Yes 🤦‍♂️

> Yes, but only with very large keys > 10_000 bytes

I don't get it. The data structure should only get deeper if the prefix of keys is equal. With uniform keys, the "risk" of needing another level is `1/256` *per level*, meaning you'd need a billion uniform keys on average to have just 5 levels.

Reply:
Yeah, it was a very specific test case to stretch the limits, definitely not a uniform keys ...

Reply:
Okay. Then it's out of scope. I won't consider call stack size an issue until proven wrong, because you would likely hit other limits (like memory) way before.